tests/utils.rs

```diff
@@ -14,12 +14,47 @@ use std::path::PathBuf;
 use test_utils::{trim_margin, TempFile};
 use yaml_rust::YamlLoader;
 
-fn compare_results(result: (ExitCode, String), expected: Result<(), &str>) {
-    let expected_output = match expected {
-        Err(expected_output) => (ExitCode(1), expected_output.to_string()),
-        Ok(()) => (ExitCode(0), "All tests passed.\n".to_string()),
-    };
-    assert_eq!(result, expected_output);
+#[derive(Debug, PartialEq)]
+pub struct Expect {
+    expected_exit_code: ExitCode,
+    expected_stdout: String,
+    expected_stderr: String,
+}
+
+impl Expect {
+    pub fn ok() -> Self {
+        Expect {
+            expected_exit_code: ExitCode(0),
+            expected_stdout: "All tests passed.\n".to_string(),
+            expected_stderr: "".to_string(),
+        }
+    }
+
+    pub fn err(expected_output: &str) -> Self {
+        Expect {
+            expected_exit_code: ExitCode(1),
+            expected_stdout: expected_output.to_string(),
+            expected_stderr: "".to_string(),
+        }
+    }
+
+    pub fn stdout(self, expected_output: &str) -> Self {
```

Comment:
since the usage is `Expect::ok().stdout(...)` my vote would be to name this in a way that suggests you're adding to the structure rather than performing some action, something like `Expect::ok().with_stdout(...)`.

Reply:
Good idea! I have made that change.